packages/mongodb-chatbot-eval/src/commands/generate.ts

```diff
@@ -1,19 +1,22 @@
 import { CommandModule } from "yargs";
-import {
-  ResolvedConfig,
-  LoadConfigArgs,
-  withConfig,
-  withConfigOptions,
-} from "../withConfig";
+import { LoadConfigArgs, withConfig, withConfigOptions } from "../withConfig";
+import { generateDataAndMetadata } from "../generate/generateDataAndMetadata";
+import { EvalConfig } from "../EvalConfig";
 
 const commandModule: CommandModule<unknown, LoadConfigArgs> = {
   command: "generate",
   builder(args) {
-    return withConfigOptions(args).string("name").demandOption("name");
+    return withConfigOptions(args)
+      .option("name", {
+        type: "string",
+        description: "Name of the data generation.",
+      })
+      .demandOption("name");
   },
   async handler(args) {
     return withConfig(generateCommand, {
       ...args,
+      name: args.name as string,
```

Comment:
typescript err if i don't:

```
Type 'unknown' is not assignable to type 'string'.ts(2322)
generate.ts(28, 15): The expected type comes from property 'name' which is declared here on type 'LoadConfigArgs & { name: string; }'
```

Reply:
Looked closer, it's because we're specifying `unknown` for the options generic (`T`) and just `LoadConfigArgs` for the args generic (`U`). This basically means that the type for `args` won't include anything we specify in the builder method.

We want this instead to avoid type assertion:

```ts
type GenerateCommandArgs = {
  name: string;
};

const commandModule: CommandModule<
  GenerateCommandArgs,
  LoadConfigArgs & GenerateCommandArgs
> = {
  command: "generate",
  builder(args) {
    return withConfigOptions(args).option("name", {
      type: "string",
      demandOption: true,
      description: "Name of the data generation.",
    });
  },
  async handler(args) {
    return withConfig(generateCommand, args);
  },
  describe: "Generate data for evaluation.",
};
```

Reply:
I have a helper function for this in the artifact generator that makes it a bit cleaner. Maybe something to include in our separate withConfig package? cc @cbush 

```ts
import { CommandModule } from "yargs";
import { LoadConfigArgs } from "./withConfig";

type CMD<T extends { [k: string]: unknown } = Record<string, never>> =
  CommandModule<unknown, LoadConfigArgs & T>;

export function createCommand<
  T extends { [k: string]: unknown } = Record<string, never>
>(commandModule: CMD<T>) {
  return commandModule;
}
```

Reply:
just did the first naive implementation to keep it simple. the typescript-foo can live in the withConfig package, and we can add it here later.